### PR TITLE
Add multi-arch image support by moving to temurin 21

### PR DESCRIPTION
**Pull request by barryoneill**
Addresses https://github.com/scala-steward-org/scala-steward/issues/2590

See: https://adoptium.net/supported-platforms/ for architecture matrix

### CODE REVIEW

**Review by mzuehlke — APPROVED** [review]
This looks good, but I am also not using the docker image. 
So I am waiting for feedback from the other maintainers before merging.

**Comment by barryoneill** [comment]
Hi @mzuehlke  - thanks for the help.  I think though there's going to be more to this.   While the contents of the image are now multi-arch friendly, I think the current CI setup is still building a single-architecture image.  I'm not familiar with the sbt-ecr plugin (I abandoned it years ago in favour of native Docker configuration), but I'll try and figure out what's needed.

**Comment by barryoneill** [comment]
oops, meant to comment on the issue.  Moving there.

**Pull request merged**
